libexec/rtld-elf/aarch64/rtld_c18n_machdep.c

```diff
@@ -137,11 +137,18 @@ tramp_compile(void **entry, const struct tramp_data *data)
 		PATCH_LDR_IMM(call_hook, function, 3);
 	}
 
+#ifdef __ARM_MORELLO_PURECAP_BENCHMARK_ABI
```

Comment:
See this is the kind of thing that concerns me, you're changing the structure of trampolines for the benchmark ABI

Reply:
I've been very careful to make sure that _most_ trampolines resemble those on the other ABI _most_ of the time so that benchmarks shouldn't be biased.

For example in this case, almost all the time `executive == false` and so it does not matter.